include/evmc/evmc.hpp

```diff
@@ -150,6 +150,108 @@ constexpr bytes32::operator bool() const noexcept
     return !is_zero(*this);
 }
 
+namespace literals
+{
+namespace internal
+{
+template <typename T, T... Ints>
+struct integer_sequence
+{
+};
+
+template <uint8_t... Bytes>
+using byte_sequence = integer_sequence<uint8_t, Bytes...>;
+
+template <char... Chars>
+using char_sequence = integer_sequence<char, Chars...>;
+
+
+template <typename, typename>
+struct concatenate;
+
+template <uint8_t... Bytes1, uint8_t... Bytes2>
+struct concatenate<byte_sequence<Bytes1...>, byte_sequence<Bytes2...>>
+{
+    using type = byte_sequence<Bytes1..., Bytes2...>;
+};
+
+
+constexpr uint8_t parse_hex_digit(uint8_t d) noexcept
+{
+    return static_cast<uint8_t>(
+        (d >= '0' && d <= '9') ? d - '0' : (d >= 'a' && d <= 'f') ? d - 'a' + 10 : d - 'A' + 10);
```

Comment:
Any chance to abort on invalid hex digits? E.g. `0x123kkx_address` ?

Reply:
This will be C++ parser error. But I've found in C++17 you can have literals like `0x0.000000000000000000000000000000000000p0_address` so I added additional checks.

Reply:
I reckon it is not easy to add a test case (e.g. file which would not compile)?

Reply:
Not without creating custom testing framework in CMake.